test/src/test/java/jenkins/model/JenkinsTest.java

```diff
@@ -471,4 +480,116 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
             throw new IOException("Something happened (the listener always throws this exception)");
         }
     }
+    
+    @Test
+    @Issue("JENKINS-39465")
+    public void agentProtocols_singleEnable_roundtrip() throws Exception {
+        final Set<String> defaultProtocols = Collections.unmodifiableSet(j.jenkins.getAgentProtocols());
+        Assume.assumeThat("We assume that JNLP3-connect is disabled", 
+                defaultProtocols, not(hasItem("JNLP3-connect")));
```

Comment:
I propose to follow-up in a separate PR. BTW, I'm pretty sure that `JNLP3-connect` will be always disabled. The test depending on JNLP4 status is more flaky